File: app/src/main/java/space/lopatkin/spb/helpboardgamecard/di/ApplicationComponent.java
package space.lopatkin.spb.helpboardgamecard.di;

import dagger.Component;
import space.lopatkin.spb.helpboardgamecard.ui.addcard.AddCardFragment;
import space.lopatkin.spb.helpboardgamecard.ui.catalog.CatalogFragment;
import space.lopatkin.spb.helpboardgamecard.ui.catalog.helpcard.cardedit.CardEditFragment;
import space.lopatkin.spb.helpboardgamecard.ui.catalog.helpcard.HelpcardFragment;
import space.lopatkin.spb.helpboardgamecard.ui.settings.SettingsFragment;

import javax.inject.Singleton;

@Singleton
@Component(modules = {
        ApplicationModule.class,
        DomainModule.class,
        DataModule.class
})
public interface ApplicationComponent {

    void inject(CatalogFragment fragment);

    void inject(AddCardFragment fragment);

    void inject(HelpcardFragment fragment);

    void inject(CardEditFragment fragment);

    void inject(SettingsFragment fragment);

}

File: app/src/main/java/space/lopatkin/spb/helpboardgamecard/di/ApplicationComponent.kt
package space.lopatkin.spb.helpboardgamecard.di

import dagger.Component
import space.lopatkin.spb.helpboardgamecard.presentation.MainActivity
import space.lopatkin.spb.helpboardgamecard.presentation.addcard.AddCardFragment
import space.lopatkin.spb.helpboardgamecard.presentation.catalog.CatalogFragment
import space.lopatkin.spb.helpboardgamecard.presentation.catalog.helpcard.HelpcardFragment
import space.lopatkin.spb.helpboardgamecard.presentation.catalog.helpcard.cardedit.CardEditFragment
import space.lopatkin.spb.helpboardgamecard.presentation.settings.SettingsFragment
import javax.inject.Singleton

@Singleton
@Component(
    modules = [
        ApplicationModule::class,
        DomainModule::class,
        DataModule::class]
)
interface ApplicationComponent {

    fun inject(activity: MainActivity)
    fun inject(fragment: CatalogFragment)
    fun inject(fragment: AddCardFragment)
    fun inject(fragment: HelpcardFragment)
    fun inject(fragment: CardEditFragment)
    fun inject(fragment: SettingsFragment)

}
